docs/source/tutorials/using_text.rst

```diff
@@ -0,0 +1,461 @@
+  works in Python. The values should be the color of the Text from :class:`~.Color`.
+  
+
+* The keys contain words or characters which should be colored separately
+  and the values should be the color from :class:`~.Color`:
+
+.. manim:: Textt2cExample
+    :save_last_frame:
+
+    class Textt2cExample(Scene):
+        def construct(self):
+            t2cindices = Text('Hello', t2c={'[1:-1]': BLUE}).move_to(LEFT)
+            t2cwords = Text('World',t2c={'rl':RED}).next_to(t2cindices, RIGHT)
+            self.add(t2cindices, t2cwords)
+
+If you want to avoid problems when using colors (due to ligatures), consider using
+:class:`MarkupText`.
+
+
+Using Gradients
+---------------
+
+You can add a gradient using :attr:`~.Text.gradient`. The value must
+be an Iterable of any length:
+
+.. manim:: GradientExample
+    :save_last_frame:
+
+    class GradientExample(Scene):
+        def construct(self):
+            t = Text("Hello", gradient=(RED, BLUE, GREEN)).scale(2)
+            self.add(t)
+
+You can also use :attr:`~.Text.t2g` for gradients with specific 
+characters of the text. It shares a similar syntax to :ref:`Using Colors`:
+
+.. manim:: t2gExample
+    :save_last_frame:
+
+    class t2gExample(Scene):
+        def construct(self):
+            t2gindices = Text(
+                'Hello',
+                t2g={
+                    '[1:-1]': (RED,GREEN),
+                },
+            ).move_to(LEFT)
+            t2gwords = Text(
+                'World',
+                t2g={
+                    'World':(RED,BLUE),
+                },
+            ).next_to(t2gindices, RIGHT)
+            self.add(t2gindices, t2gwords)
+
+Setting Line Spacing
+--------------------
+
+You can set line spacing using :attr:`~.Text.line_spacing`:
+
+.. manim:: LineSpacing
+    :save_last_frame:
+
+    class LineSpacing(Scene):
+        def construct(self):
+            a = Text("Hello\nWorld", line_spacing=1)
+            b = Text("Hello\nWorld", line_spacing=4)
+            self.add(Group(a,b).arrange(LEFT, buff=5))
+
+
+.. _disable-ligatures:
+
+Disabling Ligatures
+-------------------
+
+By disabling ligatures you would get a one-to-one mapping between characters and
+submobjects. This fixes the issues with coloring text. 
+
+
+.. warning::
+
+    Be aware that using this method with text that heavily depends on
+    ligatures (Arabic text) may yield unexpected results.
+
+You can disable ligatures by passing ``disable_ligatures`` to 
+:class:`Text`. For example:
+
+.. manim:: DisableLigature
+    :save_last_frame:
+
+    class DisableLigature(Scene):
+        def construct(self):
+            li = Text("fl ligature").scale(2)
+            nli = Text("fl ligature", disable_ligatures=True).scale(2)
+            self.add(Group(li, nli).arrange(DOWN, buff=.8))
+
+.. _iterating-text:
+
+Iterating :class:`~.Text`
+-------------------------
+
+Text objects behave like :class`~.VGroup`s. Therefore, you can slice and index
```

Comment:
This doesn't render correctly. (I usually fail to get the link syntax right in my first attempt, but it should look more or less like in the suggestion below.)

```suggestion
Text objects behave like :class:`VGroups <.VGroup>`. Therefore, you can slice and index
```

Reply:
Maybe if I'm not wrong there should be a `~`?
```suggestion
Text objects behave like :class:`VGroups <~.VGroup>`. Therefore, you can slice and index
```

Reply:
No: `~` only causes the name to be abbreviated when not specifying a custom link text. But it is possible that an underscore is necessary; not sure about that:

```suggestion
Text objects behave like :class`VGroups <.VGroup>`_. Therefore, you can slice and index
```

(I'm confident that one of our three suggestions here is correct. :-))